src/Nethermind/Nethermind.Synchronization/SnapSync/SnapSyncFeed.cs

```diff
@@ -168,18 +168,29 @@ public SyncResponseHandlingResult AnalyzeResponsePerPeer(AddRangeResult result,
                                 {
                                     if (allLastFailures == peerLastFailures)
                                     {
-                                        _logger.Trace($"SNAP - peer to be punished:{peer}");
+                                        
+                                        _snapProvider.UpdatePivot();
+                                        _resultLog.Clear();
+
+                                        _logger.Trace($"SNAP - updating pivot and retrying with peer: {peer}");
+
+                                        
+                                        return SyncResponseHandlingResult.OK;
+                                    }
+                                    else
```

Comment:
Won't this mark peer as lesser quality unless all failures were that peer? (i.e. if its less than all, it will also mark as Lesser)